SDL/CocoaBarItems.mm

```diff
@@ -0,0 +1,390 @@
+ConfigValueName = !ConfigValueName; \
+__VA_ARGS__; /* for any additional updates */ \
+item.state = [self controlStateForBool: ConfigValueName]; \
+}
+
+TOGGLE_METHOD(Sound, g_Config.bEnableSound)
+TOGGLE_METHOD(AutoFrameSkip, g_Config.bAutoFrameSkip, g_Config.UpdateAfterSettingAutoFrameSkip())
+TOGGLE_METHOD(SoftwareRendering, g_Config.bSoftwareRendering)
+TOGGLE_METHOD(FullScreen, g_Config.bFullScreen, System_MakeRequest(SystemRequestType::TOGGLE_FULLSCREEN_STATE, 0, g_Config.UseFullScreen() ? "1" : "0", "", 3))
+TOGGLE_METHOD(VSync, g_Config.bVSync)
+#undef TOGGLE_METHOD
+
+-(void)setToggleShowCounterItem: (NSMenuItem *)item {
+    [self addOrRemoveInteger:(int)(item.tag - 100) to:&g_Config.iShowStatusFlags];
+    item.state = [self controlStateForBool:g_Config.iShowStatusFlags & item.tag];
+}
+
+-(void)addOrRemoveInteger: (int)integer to: (int *)r {
+    if (integer & *r) {
+        *r -= integer;
+    } else {
+        *r |= integer;
+    }
+}
+
+-(void)setCurrentGPUBackend: (NSMenuItem *)sender {
+    std::vector<GPUBackend> allowed = [self allowedGPUBackends];
+    if (allowed.size() == 1) {
+        printf("only one item, bailing");
+        return;
+    }
+    
+    g_Config.iGPUBackend = (int)(allowed[sender.tag]);
+    sender.state = NSControlStateValueOn;
+    
+    for (NSMenuItem *item in sender.menu.itemArray) {
+        // deselect the previously selected item
+        if (item.state == NSControlStateValueOn && item.tag != sender.tag) {
+            item.state = NSControlStateValueOff;
+            break;
+        }
+    }
+}
+
+-(NSControlStateValue) controlStateForBool: (BOOL)boolValue {
+    return boolValue ? NSControlStateValueOn : NSControlStateValueOff;
+}
+
+-(std::vector<GPUBackend>)allowedGPUBackends {
+    std::vector<GPUBackend> allBackends = {
+        GPUBackend::OPENGL, GPUBackend::VULKAN,
+    };
+    
+    std::vector<GPUBackend> allowed;
+    
+    for (GPUBackend backend : allBackends) {
+        if (g_Config.IsBackendEnabled(backend)) {
+            allowed.push_back(backend);
+        }
+    }
+    
+    return allowed;
+}
+
+-(void)addOpenRecentlyItem {
+    std::vector<std::string> recentIsos = g_Config.RecentIsos();
+    NSMenuItem *openRecent = [[NSMenuItem alloc] initWithTitle:@"Open Recent" action:nil keyEquivalent:@""];
+    NSMenu *recentsMenu = [[NSMenu alloc] init];
+    if (recentIsos.empty())
+        openRecent.enabled = NO;
+    
+    for (int i = 0; i < recentIsos.size(); i++) {
+        std::string filename = Path(recentIsos[i]).GetFilename();
+        NSMenuItem *item = [[NSMenuItem alloc] initWithTitle:@(filename.c_str()) action:@selector(openRecentItem:) keyEquivalent:@""];
+        item.target = self;
+        [recentsMenu addItem:item];
+    }
+    
+    openRecent.submenu = recentsMenu;
+    [self.openMenu addItem:openRecent];
+}
+
+-(void)openRecentItem: (NSMenuItem *)item {
+    NativeMessageReceived("browse_fileSelect", g_Config.RecentIsos()[item.tag].c_str());
```

Comment:
You should be able to just use "boot" now, right?

Reply:
Addressed